src/workerd/api/pyodide/pyodide.h

```diff
@@ -68,6 +72,18 @@ class PyodideMetadataReader : public jsg::Object {
 
   int read(jsg::Lock& js, int index, int offset, kj::Array<kj::byte> buf);
 
+  bool hasMemorySnapshot() {
+    return memorySnapshot.size() > 0;
+  }
+  int getMemorySnapshotSize() {
+    return memorySnapshot.size();
+  }
+
+  void disposeMemorySnapshot() {
+    memorySnapshot = kj::heapArray<kj::byte>(0);
```

Comment:
I believe just doing `kj::mv(memorySnapshot)` may be slightly more efficient.

But really this makes me wonder why you got rid of the `kj::Maybe`. We might get a memory snapshot of size 0 at some point, it would be good to be able to disambiguate this vs. the snapshot being actually missing. And then you'll be able to just assign `kj::none` here.

Reply:
We can't have a memory snapshot of size zero, what would that even mean? The heap is currently always at least 20 mb.

Reply:
I'm thinking if one of the many layers that does the writing and storing of the memory snapshot has a bug that causes the snapshot to be set to 0 bytes.

It would just give us a clue that it's not that the snapshot isn't being generated/is missing, but rather that it's corrupted.

In general `kj::Maybe` here just seems more ideal, you can get rid of the `maybeArrayToArray` function then too.

Reply:
Well `kj::Maybe` is kind of annoying to use. I need extra branches and `KJ_IF_SOME` and stuff. My problem before was I couldn't remember how to `.unwap()` the optional. Arrays already have empty arrays as a natural "not present" value and it's more ergonomic to use that way. I guess I can do
```C++
if (val == kj::none) {
   return;
}
auto & blah = KJ_REQUIRE_NONNULL(val);
```
but there is a lot of api surface here for me to figure out.

Reply:
As it is though, even if we can distinguish "empty snapshot" from "no snapshot" we'll treat the empty snapshot as a test. It doesn't really make sense to have empty snapshots, it's not a valid value, so I think it makes the most sense to just use "empty array" as the missing value here. Makes all the code a bit shorter and I don't really see any downside.

Reply:
> Well kj::Maybe is kind of annoying to use.

It is something we use throughout the code base though, so avoiding its use makes the code less consistent with the rest of the codebase (and thus harder for others understand), and I would say is actually less ergonomic.

Plus with your change, the code now has `storedSnapshot` which is a `kj::Maybe<Array>`, but `existingSnapshot` is a `kj::Array` where a size of 0 means the snapshot is missing.

I think the extra code you'll need to handle the `kj::none` explicitly isn't particularly long.

But anyway, I don't think it's worth blocking the change over this, I just think we should in general use `kj::Maybe` in situations like this.